scipy/linalg/tests/test_lapack.py

```diff
@@ -1705,3 +1705,267 @@ def test_getc2_gesc2():
+    """
+    np.random.seed(42)
+    # set test tolerance appropriate for dtype
+    rtol = 250 * np.finfo(dtype).eps
+    atol = 100 * np.finfo(dtype).eps
+    # obtain routine
+    gtsvx = get_lapack_funcs('gtsvx', dtype=dtype)
+    # Generate random tridiagonal matrix A
+    n = 10
+    dl = generate_random_dtype_array((n-1,), dtype=dtype)
+    d = generate_random_dtype_array((n,), dtype=dtype)
+    du = generate_random_dtype_array((n-1,), dtype=dtype)
+    A = np.diag(dl, -1) + np.diag(d) + np.diag(du, 1)
+    # generate random solution x
+    x = generate_random_dtype_array((n, 2), dtype=dtype)
+    # create b from x for equation Ax=b
+    if trans == 'N':
+        b = A @ x
+    else:
+        b = (A.conj().T) @ x
+
+    # store a copy of the inputs to check they haven't been modified later
+    inputs_cpy = [dl.copy(), d.copy(), du.copy(), b.copy()]
+
+    # determine dlf, df, duf, du2f, and IPIV from lambda expression
+
+    dlf_, df_, duf_, du2f_, ipiv_, info_lambda = dl_d_du_lambda(dl, d, du)
+
+    gtsvx_out = gtsvx(dl, d, du, b, fact=fact, trans=trans, dlf=dlf_, df=df_,
+                      duf=duf_, du2=du2f_, ipiv=ipiv_)
+    dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
+    assert_(info == 0, "?gtsvx info = {}, should be zero".format(info))
+
+    # assure that inputs are unmodified
+    assert_array_equal(dl, inputs_cpy[0])
+    assert_array_equal(d, inputs_cpy[1])
+    assert_array_equal(du, inputs_cpy[2])
+    assert_array_equal(b, inputs_cpy[3])
+
+    # test that x_soln matches the expected x
+    assert_allclose(x, x_soln, rtol=rtol, atol=atol)
+
+    # assert that the outputs are of correct type or shape
+    # rcond should be a scalar
+    assert_(hasattr(rcond, "__len__") is not True,
+            "rcond should be scalar but is {}".format(rcond))
+    # ferr should be length of # of cols in x
+    assert_(ferr.shape[0] == b.shape[1], "ferr.shape is {} but shoud be {},"
+            .format(ferr.shape[0], b.shape[1]))
+    # berr should be length of # of cols in x
+    assert_(berr.shape[0] == b.shape[1], "berr.shape is {} but shoud be {},"
+            .format(berr.shape[0], b.shape[1]))
+
+
+@pytest.mark.parametrize("dtype,trans", list(zip(DTYPES, ["N"]*len(DTYPES)))
+                         + list(zip(REAL_DTYPES, ["T"]*len(REAL_DTYPES)))
+                         + list(zip(COMPLEX_DTYPES,
+                                    ["C"]*len(COMPLEX_DTYPES))))
+@pytest.mark.parametrize("fact,dl_d_du_lambda",
+                         [("F",
+                           lambda dl,
+                           d,
+                           du: get_lapack_funcs('gttrf', dtype=d.dtype)(dl,
+                                                                        d,
+                                                                        du)),
+                          ("N", lambda dl, d, du: (None, None, None, None,
+                                                   None, None))])
```

Comment:
I also think this could be neatened up a little. Defining it inside the parameter already uses quite a bit of white space, maybe if you define it as a function, as you did with `gttrf_to_lu` above 👍  will help? OTOH some people may not like having the additional top-level functions. @mdhaber any thoughts?

Reply:
Yeah that looks really ugly ; ) Let's define it outside @parametrize.

Reply:
Ok!